x/storage/keeper/keeper.go

```diff
@@ -222,15 +221,24 @@ func (k Keeper) doDeleteBucket(ctx sdk.Context, operator sdk.AccAddress, bucketI
 		return err
 	}
 	err = ctx.EventManager().EmitTypedEvents(&types.EventDeleteBucket{
-		Operator:    operator.String(),
-		Owner:       bucketInfo.Owner,
-		BucketName:  bucketInfo.BucketName,
-		BucketId:    bucketInfo.Id,
-		PrimarySpId: bucketInfo.PrimarySpId,
+		Operator:                   operator.String(),
+		Owner:                      bucketInfo.Owner,
+		BucketName:                 bucketInfo.BucketName,
+		BucketId:                   bucketInfo.Id,
+		GlobalVirtualGroupFamilyId: bucketInfo.GlobalVirtualGroupFamilyId,
 	})
 	return err
 }
 
+func (k Keeper) GetPrimarySPForBucket(ctx sdk.Context, bucketInfo *types.BucketInfo) (*sptypes.StorageProvider, error) {
+	gvgFamily, found := k.virtualGroupKeeper.GetGVGFamily(ctx, bucketInfo.GlobalVirtualGroupFamilyId)
+	if !found {
+		return nil, virtualgroupmoduletypes.ErrGVGFamilyNotExist.Wrapf("gvg family (%d) not found when force delete bucket.", bucketInfo.GlobalVirtualGroupFamilyId)
```

Comment:
The error message is not correct. By the way, it is possible to make it `MustGet...` ? Usually, we had check the existence of bucket.

Reply:
Fixed. `MustGet...` Added.